src/test/java/org/opensearch/knn/jni/JNIServiceTests.java

```diff
@@ -1134,4 +1135,150 @@ public void testCreateIndexFromTemplate() throws IOException {
         long pointer = JNIService.loadIndex(tmpFile1.toAbsolutePath().toString(), Collections.emptyMap(), FAISS_NAME);
         assertNotEquals(0, pointer);
     }
+
+    @SneakyThrows
+    public void testIndexLoad_whenStateIsShared_thenSucceed() {
```

Comment:
@jmazanec15 is there already IT for IVFPQ that are currently getting successful after this change too? If no lets add an IT.

Reply:
That makes sense. This change does not yet integrate this support for actually sharing state, so I figure this would make more sense in the next PR where we integrate the shared state into the search/warmup workflow

Reply:
sure